### ISSUE TITLE
Joroto X4S resistance and watts

### ISSUE THREAD

**Issue by Paxilexi** [cagnulein/qdomyos-zwift#2594]
Hi,
I just bought your app on iOS to use with my Joroto X4S. Unfortunately it shows for the resistance “nan” and for the watts “0”.
Is there any chance you could check the debug and maybe help me with a fix. That would be much appreciated!
Thank you
[debug-Do__Sept__12_18_25_59_2024.log](https://github.com/user-attachments/files/16981300/debug-Do__Sept__12_18_25_59_2024.log)

**Comment by cagnulein** [comment]
Hi i guess i already fixed it in the beta, 
send me an email to roberto.viola83 at gmail.com indicating this ticket and if you are on android or iOS. Thanks

**Comment by Paxilexi** [comment]
Thank you. It’s displaying values now. Unfortunately it doesn’t make a difference what kind of resistance level I set on my bike, the resulting watts are the same, no matter what resistance.

**Comment by Paxilexi** [comment]
Sure. 

[debug-Do__Sept__12_19_47_30_2024.log](https://github.com/user-attachments/files/16983109/debug-Do__Sept__12_19_47_30_2024.log)

So if I add a HR sensor it will calculate the wattage correctly? But the resistance level won’t change / passed through?

Thank you for your help, I appreciate it!

**Comment by cagnulein** [comment]
yes unfortunately the bike just sends the cadence but qz will auto
calculate power with hr and cadence and you can also calibrate the behavior
Let me know if you have any other questions

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 12 set 2024 alle 19:03 Paxilexi ***@***.***> ha
scritto:

**Comment by Paxilexi** [comment]
Damn… is there anything I could try regarding the resistance/ power? Both are shown on the bikes display.

Also apparently with the older version of the bike ( X2 pro ) it’s working.

**Comment by cagnulein** [comment]
on the joroto app are you able to get the resistance?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 12 set 2024 alle 19:22 Paxilexi ***@***.***> ha
scritto:

**Comment by Paxilexi** [comment]
You mean on the display / bike computer? Yes it shows the current resistance level ( 1-100 % ) when I’m connected to QZ ( or any other app ).
If not connected to an app it shows also wattage and the rest.

**Comment by cagnulein** [comment]
no i mean the joroto app if exist any

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 12 set 2024 alle 19:37 Paxilexi ***@***.***> ha
scritto:

**Comment by Paxilexi** [comment]
No, there is no app by Joroto. The only app that they mention is kinomap ( but i didn’t try it yet, so can’t tell if it shows something there ).

**Comment by cagnulein** [comment]
you can try kinomap but i guess it does exactly what im doing
let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 12 set 2024 alle 19:42 Paxilexi ***@***.***> ha
scritto:
